lnwire/lnwire.go

```diff
@@ -171,12 +171,35 @@ func writeElement(w io.Writer, element interface{}) error {
 			}
 		}
 	case *btcec.Signature:
+		// Serialize the signature with all the checks that entails
 		sig := e.Serialize()
-		if len(sig) > 73 {
-			return fmt.Errorf("Signature too long!")
+		var b [64]byte
+		// Extract lengths of R and S
+		rLen := uint8(sig[3])
+		sLen := uint8(sig[5+rLen])
+		// Check to make sure each can fit into the intended buffer
+		if sLen > 32 {
+			if (sLen > 33) || (sig[6+rLen] != 0x00) {
+				return fmt.Errorf("S is over 32 bytes long without padding")
+			} else {
+				sLen -= 1
+				copy(b[64-sLen:], sig[7+rLen:])
+			}
+		} else {
+			copy(b[64-sLen:], sig[6+rLen:])
+		}
+		if rLen > 32 {
```

Comment:
R needs to be after S because we subtract one from rLen, which affects the index lookups in the S extraction. I can create new variables to track this but it wouldn't be as efficient. Let me know what you prefer.